### ISSUE TITLE
Detect path join with hardcoded absolute path

### ISSUE THREAD

**Issue by joshtriplett** [rust-lang/rust-clippy#10655]
### What it does

Detect calls to `.join` with an absolute path as an argument. This won't do what the user likely expects. `join` with an absolute path will ignore the left-hand side and just use the absolute path.

For example, `root.join("/bin/sh")` will return the path `"/bin/sh"`.

In an ideal world, this would also detect calls like `root.join(SOME_CONSTANT)` where the constant is a hardcoded absolute path, as well, but that may be harder. This would still provide value even if it only detects absolute paths specified directly in the `join` call.

The lint should explain the behavior of `join`, and should suggest removing the leading `/` from the absolute path.

### Lint Name

join_absolute_path

### Category

correctness

### Advantage

Removing the leading `/` from the user's path will make the join create a path relative to the path given on the left-hand side of the join operation.

### Example

```rust
some_path.join("/bin/sh")
```

Could be written as:

```rust
some_path.join("bin/sh")
```

**Comment by xFrednet** [comment]
Hey @anthonygedeon, there is already an open PR https://github.com/rust-lang/rust-clippy/pull/10663 to implement the lint. This is just a quick warning to avoid duplicate work :)

**Comment by anthonygedeon** [comment]
@xFrednet Thanks for letting me know! I'll try to look for another good-first-issue then
